.github/workflows/macos.yml

```diff
@@ -0,0 +1,270 @@
+    runs-on: ${{ matrix.os }}
+    env:
+      GIT_LFS_SKIP_SMUDGE: 1
+      HOMEBREW_NO_AUTO_UPDATE: 1
+      APPLE_ID: ${{ secrets.APPLE_ID }}
+      DEVELOPMENT_TEAM: ${{ secrets.APPLE_DEVELOPMENT_TEAM }}
+      PKG_SIGN_IDENTITY: ${{ secrets.APPLE_PKG_SIGN_IDENTITY }}
+      CODE_SIGN_IDENTITY: ${{ secrets.APPLE_CODE_SIGN_IDENTITY }}
+
+      BUILD_DIR: build
+      REL_DESC_FILE: "build/rel_desc.md"
+
+      # Map test matrix elements to environmental variables, so that scripts launched from CI scripts may react to them.
+      # Some matrix variables have defaults if unspecified by the matrix
+      CMAKE_GEN: ${{ matrix.generator }}
+      ARCH: ${{ matrix.arch }} # iOS uses ARCH
+      ARCHS: ${{ matrix.arch }} # MacOS uses ARCHS
+      CONFIGURATION: ${{ matrix.options.config }}
+      PLATFORM: ${{ matrix.options.platform }}
+      FEATURE_DOC: ${{ matrix.options.doc }}
+      FEATURE_JNI: ${{ matrix.options.jni }}
+      FEATURE_LOADTESTS: ${{ matrix.options.loadtests }}
+      FEATURE_PY: ${{ matrix.options.py }}
+      FEATURE_TESTS: ${{ matrix.options.tests && matrix.options.tests || 'ON' }}
+      FEATURE_TOOLS: ${{ matrix.options.tools }}
+      FEATURE_TOOLS_CTS: ${{ matrix.options.tools_cts }}
+      PACKAGE: ${{ matrix.options.package && matrix.options.package || 'NO' }}
+      SUPPORT_OPENCL: ${{ matrix.options.opencl }}
+      SUPPORT_SSE: ${{ matrix.options.sse }}
+      WASM_BUILD: ${{ matrix.options.wasm && matrix.options.wasm || 'OFF' }}
+      WERROR: ${{ matrix.options.werror && matrix.options.werror || 'ON' }}
+      # CC Handled by job step
+      # CXX Handled by job step
+
+      PYTHON_DIST_DIR: interface/python_binding/dist
+
+      VULKAN_SDK_VER: ${{ matrix.vk_sdk_ver }}
+    # This is just to tell the Vulkan install script where to install.
+    # NOTE: Hardcoded $HOME expansion, due to 'env' context being unavailble in env:
+      VULKAN_INSTALL_DIR: /Users/runner/VulkanSDK/${{ matrix.vk_sdk_ver }}
+      VULKAN_SDK: /Users/runner/VulkanSDK/${{ matrix.vk_sdk_ver }}/macOS
+
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          # Fetch all history to make sure tags are
+          # included (used for version creation)
+          fetch-depth: 0
+
+      - name: before_install
+        run: |
+          echo -n "Running on the following GitHub Actions CI runner: " && uname -a
+          echo -n "CMake version on the runner is " && cmake --version
+          echo -e "machine github.com\n  login $GITHUB_TOKEN" >> ~/.netrc # Prevent rate limiting on Git LFS.
+          cat ~/.netrc
+          if [[ -n "$FEATURE_LOADTESTS" && "$FEATURE_LOADTESTS" != "OFF" ]]; then
+            # Install vcpkg.
+            pushd $HOME
+            git clone https://github.com/microsoft/vcpkg
+            cd vcpkg && ./bootstrap-vcpkg.sh -disableMetrics && echo "VCPKG_ROOT=`pwd`" >> $GITHUB_ENV
+            popd
+          fi
+
+      - name: Install Doxygen
```

Comment:
I noticed this and Install Graphviz are running in the iOS job. I think you need

```
if: matrix.options.doc == 'ON'
```

Reply:
Nice catch. Fixed.